src/builder.rs

```diff
@@ -57,4 +61,34 @@ impl Builder {
                 .and_then(|obj| obj.downcast().ok())
         }
     }
+
+    pub fn add_from_file<T: AsRef<Path>>(&self, file_path: T) -> Result<bool, Error> {
```

Comment:
I was wrong about failing, if ffi function accept error we must return it.
But we return unit `()` in good case instead `bool`.